src/coreclr/nativeaot/Runtime/eventpipe/ep-rt-aot.cpp

```diff
@@ -85,12 +84,66 @@ ep_rt_aot_sample_profiler_write_sampling_event_for_threads (
 {
 }
 
+#if defined (__linux__)
+extern char *__progname;
+#endif
+
 const ep_char8_t *
 ep_rt_aot_entrypoint_assembly_name_get_utf8 (void) 
 {
-    // shipping criteria: no EVENTPIPE-NATIVEAOT-TODO left in the codebase
-    // TODO: Implement EventPipe assembly name - return filename in nativeaot?
-    return reinterpret_cast<const ep_char8_t*>("");
+    // Although PalGetModuleFileName exists, non-windows implementation has issues and we use __progname for Linux.
```

Comment:
What are the issues - is there something tracking fixing it? Presumably other things are using it - how are they affected?

Reply:
> non-windows implementation has issues and we use __progname for Linux.

Is this referring to the comments under `#if HOST_WASM` in PalGetModuleFileName implementation? We don't care about EventPipe for WASM at this point and it's unclear if `__progname` would work there either.

Reply:
Linux return null for this call when the 2nd parameter is null. Haven't been able to figure out how to get the handle for the module to make the call successfully.

Reply:
> Haven't been able to figure out how to get the handle for the module to make the call successfully.

Did you look at the existing uses of `PalGetModuleFileName`? `PalGetModuleHandleFromPointer(&ep_rt_aot_entrypoint_assembly_name_get_utf8)` should give a good handle and would likely be more correct for the "NativeAOT built into a shared library" case.